tests/basics/decorator_full.py

```diff
@@ -0,0 +1,71 @@
+### Test decorator syntax.
+
+###############################################################################
+### Test for syntax errors.
+###############################################################################
+def test_for_syntax_error(code):
+    try:
+        exec(code)
+    except SyntaxError:
+        print("SyntaxError")
+
+decs = [
+    "x,",
+    "x, y",
+    "x = y",
+    "pass",
+    "import sys",
+    "@dec"
+]
+
+for dec in decs:
+    code = f"@{dec}\ndef foo(): pass"
+    test_for_syntax_error(code)
+
+###############################################################################
+### Test for type errors (previously syntax errors before Python 3.9).
+###############################################################################
+def test_for_type_error(code):
+    try:
+        exec(code)
+    except TypeError:
+        print("TypeError")
+
+decs = [
+    "1.+2j",
```

Comment:
Can this test be modified so it runs on miminal targets, ie so that there doesn't need to be special handling in the test suite runners to exclude this full decorator test from certain ports?

Can this complex number be replaced with something else (or just removed)?  Is it testing something (providing test coverage) that can't be tested in any other way?  If so it would be good to move just this complex-number based test to another file and put it in the `tests/float/` directory so it's automatically skipped.

Reply:
I removed the complex number test and cleaned up the test runners in the latest commit.